File: sem2/po/po_proj2/src/studia/prog_ob/projekt2/world_simulator/organisms/animals/animals_subclasses/Sheep.java
package studia.prog_ob.projekt2.world_simulator.organisms.animals.animals_subclasses;


import studia.prog_ob.projekt2.world_simulator.World;
import studia.prog_ob.projekt2.world_simulator.maps.MapLocation;
import studia.prog_ob.projekt2.world_simulator.organisms.Organism;
import studia.prog_ob.projekt2.world_simulator.organisms.animals.Animal;

public class Sheep extends Animal {

    public static final int SHEEP_STRENGTH = 4;
    public static final int SHEEP_INITIATIVE = 4;
    public static final int SHEEP_SPEED = BASE_ANIMAL_SPEED;
    public static final char SHEEP_SYMBOL = 'S';

    public Sheep(MapLocation new_map_location, World new_world) {
        super(SHEEP_STRENGTH, SHEEP_INITIATIVE, new_map_location, SHEEP_SYMBOL, new_world, SHEEP_SPEED);
    }

    @Override
    public Organism spawn_offspring(MapLocation location) {
        return getWorld().add_organism(new Sheep(location, getWorld()));
    }

    @Override
    public String toString() {
        return "Sheep " + getMap_location() + " ";
    }
}

File: sem2/po/PO-VirtualWorld-Python/simulator/organisms/animals/Sheep.py
from simulator.organisms.animals.Animal import Animal, BASE_ANIMAL_SPEED
from simulator.organisms.Organism import ORGANISM_TYPES

SHEEP_STRENGTH = 4
SHEEP_INITIATIVE = 4
SHEEP_SPEED = BASE_ANIMAL_SPEED



class Sheep(Animal):
    def __init__(self, new_world, new_location):
        super().__init__(new_world, new_location, SHEEP_STRENGTH, SHEEP_INITIATIVE, ORGANISM_TYPES["Sheep"], SHEEP_SPEED)

    def spawn_offspring(self, new_location):
        return Sheep(self.world, new_location)

    def __str__(self):
        return "Sheep at " + str(self.location)
